### DISCUSSION TITLE
How to create two separate API Documentation on one FastAPI system?

### DISCUSSION THREAD

**Discussion by richardridwan20** [fastapi/fastapi#6840] [Questions] [answered]
Hi, currently I'm working with FastAPI to create a microservice that can be used for both internally and externally. The thing is I can't figure out how to create separate documentation for FastAPI.

For example:
1. Internal Documentation contains all the routes for internal developers to test
2. External Documentation only contains certain routes for external developers to try

Any help or feedbacks would be appreciated, thankyou

**Comment by PrasanthChettri** [comment]
@richardridwan20 I usually do this by making two different FastApi classes. I think you could do something like
```
  int_app = FastAPI()
  ext_app = FastAPI()
```
and mounting the needed routers, ofcourse this has it's own drawbacks, if there is a better way, it would be nice

**Comment by ycd** [comment]
You can copy the original `openapi.json` then you can serve it in a different route without creating two different FastAPI instance.

**Comment by richardridwan20** [comment, marked as answer]
@PrasanthChettri @ycd Thankyou for the help! 

I eventually managed to create another Sub Application and mount it to the Main one, then create another Router to accomodate the needs.

For reference:
https://fastapi.tiangolo.com/advanced/sub-applications/

**Reply by lodwig** [reply]
Hi, i got the same problem with you and thank you  for your answer, but can you help me how you manage the middleware system i have different headers from internal and external.  i use jwt for the internal but for external i used RSA

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for reporting back and closing the issue @richardridwan20  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
